corehq/apps/reports/tests/test_views.py

```diff
@@ -0,0 +1,578 @@
+        # NOTE: user permissions get cached, so if these permissions
+        # were changed between checks, the cache would need to be cleared
+        self.reports_role.permissions = Permissions(view_report_list=list(report_names))
+        self.reports_role.save()
+
+    def _generate_report(self, export_id=None, report_name='test_report', domain=None, content=b'Some File'):
+        export_id = export_id or self.export_id
+        domain = domain or self.domain
+        file = io.BytesIO(content)
+        self.db.put(file,
+            domain=domain,
+            name='MyTestExport',
+            parent_id=export_id,
+            key=export_id,
+            type_code=1,
+            timeout=300,
+            properties={'report_class': report_name})
+        self.addCleanup(self.db.delete, export_id)
+
+
+class TestViewScheduledReport(TestCase):
+    def test_missing_report_returns_404(self):
+        with self.assertRaises(Http404):
+            self.view_scheduled_report('nonexistant_id')
+
+    def test_mismatched_domains_returns_404(self):
+        report = self._create_scheduled_report(domain='not_your_domain')
+
+        with self.assertRaises(Http404):
+            self.view_scheduled_report(report._id, domain=self.domain)
+
+    def test_user_with_no_report_access_receives_404(self):
+        self._set_user_report_access()  # Ensure the user has access to no reports
+        # User should be unable to view even his own report
+        scheduled_report = self._create_scheduled_report(owner_id=self.user._id)
+
+        with self.assertRaises(Http404):
+            self.view_scheduled_report(scheduled_report._id)
+
+    def test_user_without_scheduled_report_access_recieves_404(self):
+        scheduled_report = self._create_scheduled_report(owner_id='not_you')
+
+        with self.assertRaises(Http404):
+            self.view_scheduled_report(scheduled_report._id)
+
+    def test_user_without_permissions_receives_error_message(self):
+        saved_report = self._create_saved_report(user_id='not_you')
+        scheduled_report = self._create_scheduled_report(config_ids=[saved_report._id])
+
+        with self.assertRaises(Http404):
+            self.view_scheduled_report(scheduled_report._id)
+
+    @unittest.skip
+    def test_can_view_scheduled_report(self):
+        # TODO: Likely need to look into `send_to_elasticsarch`, especially in some existing
+        # tests, to figure out how to populate elasticsearch so that the resulting report
+        # can be generated
+        saved_report = self._create_saved_report(slug='worker_activity')
+        scheduled_report = self._create_scheduled_report(config_ids=[saved_report._id])
+
+        self._set_user_report_access(REPORT_NAME_LOOKUP['worker_activity'])
+
+        response = self.view_scheduled_report(scheduled_report._id)
+        self.assertEqual(response.status_code, 200)
+
+# ################### Helpers / Setup
+
+    def view_scheduled_report(self, report_id, domain=None):
```

Comment:
I like the pattern of putting tests first and helper and setup methods at the end. Nicely readable.